### PR TITLE
Add a quick-start guide

### PR DESCRIPTION
**Pull request by PeterRugg**
Here's the first cut at a quick start guide (with help from @gameboo). I'm sure I'll have missed important things, so feel free to comment liberally and I'll do my best to make quick fixes to get this into the release!

Fixes https://github.com/riscv/riscv-cheri/issues/339

### CODE REVIEW

File: src/summary.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+A CHERI core adds state to allow capabilities to be used from within registers, and to ensure they are not corrupted as they flow through the system.
+This means the following state is added:
+
+* Metadata within architectural registers: `XLEN`-wide integer registers (e.g. `sp`, `a0`) are all extended with another `XLEN` bits of capability metadata, including bounds and permissions.
+  The resulting `CLEN` bits in full form a capability, and we refer to the same register prefixed with a `c`, i.e. `csp`, `ca0`.
+  The integer part of the register is interpreted as the address field of the capability.
+  The zero register is extended with zero metadata and a cleared tag: this is called the *null* capability.
+  As well as general purpose registers, system registers that store addresses are extended to contain capabilities.
+  For example, `tvec` is extended to a capability version `tcc` (*trap code capability*) to allow the code bounds to be changed on an exception.
```
Comment (tariqkurd-repo) [inline src/summary.adoc:36]: Xtvec, not tvec, and we now suffix a c e.g. mtvec becomes mtvecc

File: src/summary.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+** The tags are also tracked through the memory subsystem: every aligned `CLEN`-bits wide region has a non-addressable one-bit tag, which the hardware manages atomically with the data.
+   The tag is cleared if the memory region is ever written other than using a capability store from a tagged capability register.
+   Any caches must preserve this abstraction.
+
+=== Checking Memory
+
+Every memory access performed by a CHERI core must be authorised by a capability.
+It is explicitly defined for every instruction where to find the capability to check against.
+In *purecap* code, the capability and address are used together, so e.g. `lw t0, 16(csp)` loads a word from memory, getting the address and bounds from the `csp` register.
```
Comment (tariqkurd-repo) [inline src/summary.adoc:51]: can you define `purecap` here?

File: src/summary.adoc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+For code that has not yet been fully adapted to CHERI (*hybrid* code), the processor can run in a pointer mode (not to be confused with a privilege mode) where the authorising capability is instead taken from a special CSR: the *Default Data Capability* (`DDC`)
+
+Instruction fetch is also authorised by a capability: the *Program Counter Capability* (`PCC`) which extends `PC`.
+This allows code fetch to be bounded, preventing a wide range of attacks that subvert control flow with integer data.
+Where {cheri_default_ext_name} is supported, the `PCC` also contains the mode bit indicating whether the processor is running in integer or capability pointer mode.
+Changing the bounds used for instruction fetch or the pointer mode can be as easy as performing a capability-based jump (`cjalr`).
```
Comment (tariqkurd-repo) [inline src/summary.adoc:57]: not cjalr anymore!

**Comment by tariqkurd-repo** [comment]
Also, can you make things like instructions and CSR references into cross references as they are in the rest of the spec?

**Comment by PeterRugg** [comment]
> Also, can you make things like instructions and CSR references into cross references as they are in the rest of the spec?

Hopefully fixed

File: src/summary.adoc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+The added instructions can be split into the following categories:
+
+* Capability manipulations (e.g. <<CADD>>, <<SCBNDS>>): for security, capabilities can only be modified in restricted ways.
+  Special instructions are provided to perform these allowed operations, for example _shrinking_ the bounds or _reducing_ the permissions.
+  Any attempt to manipulate capabilities without using the instructions clears the tag, rendering them unusable for accessing memory.
+
+* Capability inspection (e.g. <<GCBASE>>, <<GCPERM>>): capability fields (for example the _bounds_ describing what addresses the capability gives access to) are stored compressed in registers and memory.
+  These instructions give convenient access to allow software to query them.
+
+* Memory access instructions (e.g. `lc`, `sc`): capabilities must be read from and written to memory atomically along with their tag.
```
Comment (tariqkurd-repo) [inline src/summary.adoc:74]:
```suggestion
* Memory access instructions (e.g. <<lc>>, <<sc>>): capabilities must be read from and written to memory atomically along with their tag.
```

Comment (PeterRugg) [reply src/summary.adoc]: Yep, @gameboo just spotted them too. Hopefully fixed!

File: src/summary.adoc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+=== Added State
+
+A CHERI core adds state to allow capabilities to be used from within registers, and to ensure they are not corrupted as they flow through the system.
+This means the following state is added:
+
+* Metadata within architectural registers: XLEN-wide integer registers (e.g. `sp`, `a0`) are all extended with another XLEN bits of capability metadata, including bounds and permissions.
+  The resulting CLEN bits in full form a capability, and we refer to the same register prefixed with a `c`, i.e. `csp`, `ca0`.
+  The integer part of the register is interpreted as the address field of the capability.
+  The zero register is extended with zero metadata and a cleared tag: this is called the <null-cap> capability.
```
Comment (andresag01) [inline src/summary.adoc:34]:
```suggestion
  The zero register is extended with zero metadata and a cleared tag: this is called the <<null-cap>> capability.
```

Comment (PeterRugg) [reply src/summary.adoc]: Thanks!

File: src/summary.adoc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+** The tags are also tracked through the memory subsystem: every aligned CLEN-bits wide region has a non-addressable one-bit tag, which the hardware manages atomically with the data.
+   The tag is cleared if the memory region is ever written other than using a capability store from a tagged capability register.
+   Any caches must preserve this abstraction.
+
+=== Checking Memory
+
+Every memory access performed by a CHERI core must be authorised by a capability.
+It is explicitly defined for every instruction where to find the capability to check against.
+In _purecap_ code, where all pointers are individual capabilities, the capability and address are used together, so e.g. `lw t0, 16(csp)` loads a word from memory, getting the address and bounds from the `csp` register.
+For code that has not yet been fully adapted to CHERI (_hybrid_ code), the processor can run in a pointer mode (not to be confused with a privilege mode) where the authorising capability is instead taken from a special CSR: the default data capability (<<ddc>>)
```
Comment (andresag01) [inline src/summary.adoc:52]:
```suggestion
For code that has not yet been fully adapted to CHERI (_hybrid_ code), the processor can run in a pointer mode (not to be confused with a privilege mode) where the authorising capability is instead taken from a special CSR: the default data capability (<<ddc>>).
```

File: src/summary.adoc [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+Instruction fetch is also authorised by a capability: the program counter capability (<<pcc>>) which extends PC.
+This allows code fetch to be bounded, preventing a wide range of attacks that subvert control flow with integer data.
+Where {cheri_default_ext_name} is supported, the <<pcc>> also contains the mode bit indicating whether the processor is running in integer or capability pointer mode.
```
Comment (andresag01) [inline src/summary.adoc:56]:
```suggestion
Where {cheri_default_ext_name} is supported, the <<pcc>> also contains the <<m_bit,mode bit>> indicating whether the processor is running in integer or capability pointer mode.
```

**Review by andresag01 — APPROVED** [review]
Thanks for the changes! Just a minor adjustment

File: src/summary.adoc (hunk 2 above)
Comment (andresag01) [reply src/summary.adoc]: Thanks. Looks good!

**Pull request merged**